src/Version10/Definition/Builder/ConsoleOptionBuilderInterface.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phpcq\PluginApi\Version10\Definition\Builder;
+
+/**
+ * This interface describes a command option.
+ */
+interface ConsoleOptionBuilderInterface
+{
+    public const VALUE_SEPARATOR_SPACE = ' ';
+
+    public const VALUE_SEPARATOR_EQUAL_SIGN = '=';
+
+    public const VALUE_SEPARATOR_NONE = '';
+
+    /**
+     * Mark the option as required. If not defined the option is considered as optional.
+     *
+     * @return self
+     */
+    public function isRequired(): self;
+
+    /**
+     * Mark the option as an array if the option may occur multiple times.
+     *
+     * @return self
+     */
+    public function isArray(): self;
+
+    /**
+     * Declare that a value is required for this option.
+     *
+     * @return self
+     */
+    public function withRequiredValue(): self;
+
+    /**
+     * Declare a shortcut which may be used as an alias for the option.
+     *
+     * @return $this
+     */
+    public function withShortcut(string $shortcut): self;
+
+    /**
+     * Use the name as shortcut and do not provide a full name option like --name, instead -name is available.
+     *
+     * @return $this
+     */
+    public function withShortcutOnly(): self;
+
+    /**
+     * Declare that a value is optional for this option. A default value may be defined also.
+     *
+     * If multiple option values are supported, the method may be called multiple times.
```

Comment:
Is mutually exclusive with `ConsoleOptionBuilderInterface::withRequiredValue()` I suppose?

What exactly is an optional named value? What is the name?

Can you clarify the sentence:
> If multiple option values are supported, the method may be called multiple times.

If I do:
```php
/** @var ConsoleCommandBuilderInterface $commandBuilder */

$builder = $commandBuilder->describeOption('foo', 'foo is the base of all examples!');

/** @var ConsoleOptionBuilderInterface $builder */
$builder
    ->withOptionalValue('opt1', 'bar1')
    ->withOptionalValue('opt2', 'bar2')
    ->withOptionalValue('opt3', 'bar3');
```
What does this then mean?
What will the console command arguments then look like?
I think I need a reference implementation or some pseudo code to understand this.

Reply:
> What exactly is an optional named value? What is the name?

Some tools uses a name differs from the option to describe the option value. To automate the help translation from the tool to phpcq I want to allow to define the name of the option value.

Psalm uses has an option where multiple option values are supported ` -i, --init [source_dir=src] [level=3]`, that's why you may be able to call this method multiple times.

> What does this then mean?
> What will the console command arguments then look like?

`---foo [opt1=bar1] [opt2=bar2] [opt3=bar3]`